File: android/src/main/java/com/test/plugins/popup/PopupPluginPlugin.java
package com.test.plugins.popup;

import android.widget.Toast;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PopupPlugin")
public class PopupPluginPlugin extends Plugin {

    private PopupPlugin implementation = new PopupPlugin();

    @PluginMethod
    public void echo(PluginCall call) {
      String value = call.getString("value");

      JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }

    @PluginMethod
    public void showPopup(PluginCall call) {
      String messageText = call.getString("message");

      Toast.makeText(getActivity(), messageText, Toast.LENGTH_LONG).show();

      call.resolve();
    }
}

File: ios/Plugin/PopupPluginPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(PopupPluginPlugin)
public class PopupPluginPlugin: CAPPlugin {
    private let implementation = PopupPlugin()

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation.echo(value)
        ])
    }
}
